### ISSUE TITLE
slimdom.DocumentFragment as context or variable

### ISSUE THREAD

**Issue by egh** [FontoXML/fontoxpath#643]
When I pass in a `slimdom.DocumentFragment` as the context or as the value of a variable, it doesn't seem to work correctly when converting to a string. I'm not sure if this is expected to work or not. I can't find anything in the spec about document fragments. So I'm reporting here, looking for more information.

```
const slimdom =  require("slimdom")
const fontoxpath = require ("fontoxpath")

// Create the document <foo>Test</foo> and convert to string
let d = new slimdom.Document();
d.appendChild(d.createElement("foo"))
d.documentElement.appendChild(new slimdom.Text("Test"))
console.log(fontoxpath.evaluateXPathToString("string($d)", undefined, undefined, {d: d}));
console.log(fontoxpath.evaluateXPathToString("string(.)", d, undefined));

// Now try with a document fragment
let f = d.createDocumentFragment();
f.appendChild(d.createElement("foo"))
f.firstChild.appendChild(new slimdom.Text("Test"))
console.log(fontoxpath.evaluateXPathToString("string($f)", undefined, undefined, {f: f}));
console.log(fontoxpath.evaluateXPathToString("string(.)", f, undefined));
```

Expected results: 
```
Test
Test
Test
Test
```

Actual results:
```
Test
Test
```

Interestingly, if I call `console.log(fontoxpath.evaluateXPathToNodes(".", f, undefined))`, it returns the node, so something seems to be happening here...

**Comment by DrRataplan** [comment]
Hey @egh,

Oof. That was never tested. Evaluating to a string(either with a `string()` function or with `evaluateXPathToString` triggers the atomize algorithm. Thattakes all text descendants and append them to a string.

* The algorithm is here: https://github.com/FontoXML/fontoxpath/blob/35d0da50bcbd81f905382c5731d28ff2522968c7/src/expressions/dataTypes/atomize.ts#L19. This is missing a case for `DocumentFragment`. It should behave the same as a `Document` I think.
* The dom facade seems to return just child nodes, regardless of the NodeType of the argument. That should work: https://github.com/FontoXML/fontoxpath/blob/master/src/domFacade/ExternalDomFacade.ts#L25


Interesting case!

**Comment by egh** [comment]
Oh, should be easy enough to fix then. I'll try to file a PR. Thank you for the help!

**Comment by egh** [comment]
fixed by #644
